pints/_mcmc/_slice_covariance_matching.py

```diff
@@ -0,0 +1,354 @@
+        c_bar = np.dot(np.dot(np.linalg.inv(self._R), np.transpose(
+            np.linalg.inv(self._R))), self._c_bar_star)
+
+        # Draw second p-variate
+        z = np.random.multivariate_normal(mean, cov)
+
+        # Draw sample
+        self._proposed = c_bar + np.dot(np.linalg.inv(self._R), z)
+
+        # Set flag indicating we have created a new proposal. This is used to
+        # store the value of the log_pdf of the proposed point in the tell()
+        # method
+        self._sent_proposal = True
+
+        # Set flag that will be used if the proposal is rejected
+        self._calculate_fx_u = True
+
+        # Send trial point for checks
+        self._ready_for_tell = True
+        return np.array(self._proposed, copy=True)
+
+    def current_log_pdf(self):
+        """ See :meth:`SingleChainMCMC.current_log_pdf()`. """
+        return np.copy(self._current_log_pdf)
+
+    def current_slice_height(self):
+        """
+        Returns current log_y used to define the current slice.
+        """
+        return self._current_log_y
+
+    def name(self):
+        """ See :meth:`pints.MCMCSampler.name()`. """
+        return 'Slice Sampling - Covariance-Adaptive: Covariance Matching'
+
+    def n_hyper_parameters(self):
+        """ See :meth:`TunableMethod.n_hyper_parameters()`. """
+        return 2
+
+    def needs_sensitivities(self):
+        """ See :meth:`pints.MCMCSampler.needs_sensitivities()`. """
+        return True
+
+    def set_hyper_parameters(self, x):
+        """
+        The hyper-parameter vector is ``[sigma_c, theta]``.
+        See :meth:`TunableMethod.set_hyper_parameters()`.
+        """
+        self.set_sigma_c(x[0])
+        self.set_theta(x[1])
+
+    def set_sigma_c(self, sigma_c):
+        """
+        Sets standard deviation of initial crumb.
+        """
+        sigma_c = float(sigma_c)
+        if sigma_c < 0:
+            raise ValueError('Inital crumb standard deviation'
+                             'must be positive.')
+        self._sigma_c = sigma_c
+
+    def set_theta(self, theta):
+        """
+        Sets parameter to control how fast the precision of the
+        proposal distribution increases.
+        """
+        self._theta = float(theta)
+
+    def sigma_c(self):
+        """
+        Returns standard deviation of initial crumb.
+        """
+        return self._sigma_c
+
+    def tell(self, reply):
+        """ See :meth:`pints.SingleChainMCMC.tell()`. """
+
+        # Check ask/tell pattern
+        if not self._ready_for_tell:
+            raise RuntimeError('Tell called before proposal was set.')
+        self._ready_for_tell = False
+
+        # Unpack reply
+        fx, grad = reply
+
+        # Check reply, copy gradient
+        fx = float(fx)
+        grad = pints.vector(grad)
+
+        # If this is the log_pdf of a new point, save the value and use it
+        # to check ``f(x_1) >= y``
+        if self._sent_proposal:
+            self._proposed_pdf = fx
```

Comment:
I'd call this _proposed_log_pdf to make it clearer that this is a logged quantity